rs/registry/fetch_large_record_test_canister/src/main.rs

```diff
@@ -0,0 +1,48 @@
+//! This is just used by a test (to wit, rs/nervous_system/canisters/tests/registry.rs).
+//!
+//! Our job is just to call the registry_get_changes_since method. For that
+//! test, this is the code under test.
+
+use ic_cdk::update;
+use ic_crypto_sha2::Sha256;
+use ic_nervous_system_canisters::registry::{Registry, RegistryCanister};
+use ic_registry_fetch_large_record_test_canister::{
+    CallRegistryGetChangesSinceRequest, ContentSummary,
+};
+use ic_registry_transport::pb::v1::RegistryDelta;
+use ic_types::RegistryVersion;
+
+#[update]
+async fn call_registry_get_changes_since(
+    _: CallRegistryGetChangesSinceRequest,
+) -> Option<ContentSummary> {
+    let registry_canister = RegistryCanister::new();
+
+    let latest_version: RegistryVersion = registry_canister.get_latest_version().await.unwrap();
```

Comment:
This line is the cause.
The canister is using ic-cdk v0.18 (the `update` macro is from v0.18).
The `ic_nervous_system_canisters::registry::Registry` is still on ic-cdk v0.17.
`registry_canister.get_latest_version()` is making an inter-canister call v0.17.
It cannot be properly executed by the v0.18 executor.

For now, you can change the `ic-cdk` dependency of this crate to 0.17.1 (also sync in BUILD.bazel).
The change in `rs/nervous_system/canisters/src/registry.rs` couldn't solve the issue and caused more errors in other tests.

Reply:
🙏 Done!